glyphs_plist/src/plist.rs

```diff
@@ -414,6 +440,51 @@ impl<'a> Token<'a> {
     }
 }
 
+impl<'a> UnicodeEscapeLexer<'a> {
+    // \UXXXX
+    const ESCAPE_LEN: usize = 6;
+
+    fn new(slice: &'a str) -> Self {
+        // Not strictly required, but you're probably holding it wrong if this
+        // isn't the case
+        debug_assert!(
+            slice.starts_with(r"\U"),
+            "UnicodeEscapeLexer: slice should start with \\U"
+        );
+        Self { slice, index: 0 }
+    }
+
+    fn remaining(&self) -> usize {
+        self.slice.len() - self.index
+    }
+
+    fn consume_escapes(mut self) -> Result<(usize, Vec<u16>), Error> {
+        let mut seq = Vec::with_capacity(1);
+        loop {
+            match self.remaining() {
+                // Not enough left to parse anything or cause any problems
+                0..2 => return Ok((self.index, seq)),
+                // Nothing relevant left
+                2.. if &self.slice[self.index..self.index + 2] != r"\U" => {
+                    return Ok((self.index, seq));
+                },
+                // Not enough hex digits after the \U
+                2..Self::ESCAPE_LEN => return Err(Error::InvalidUnicode),
+                // All good, advance and try to parse something
+                Self::ESCAPE_LEN.. => {
+                    self.index += 2;
+                    let nibble = &self.slice[self.index..self.index + 4];
```

Comment:
giganit: nibble means four bits, so this should be 'nibbles' as there's four bits encoded in each hexadecimal char